### PR TITLE
Feat: like button

### LINKED ISSUE #513: implement "Like" button
Like button has two states: 
image
image

find icons to download [here](https://xd.adobe.com/view/263c33d4-4ee4-441b-7c68-f4533c94f5ee-500d/screen/bdf48a4e-501b-482a-a052-24f94e9e1baa)

### LINKED ISSUE #745: Make sure project image scales properly on project page

### PR DESCRIPTION
**Pull request by verena-e-p**
Closes #513 
Closes #745 

# Description #745 #

When the pictures of the project page were scaled down (due to decreasing in the window width) they lost their original aspect ratio. This is now fixed.

# Description #513 #

The LikeButton component is now implemented and added to the project page as well as a link to show all users who have liked the project. It is implemented very similarly to the FollowButton.

## Design ##
* different designs for different screen sizes
* different designs to indicate whether a user is already liking a project or not

## Functionality Button ##
* new backend model to store the likes
* check whether a user is liking
* set/remove a like
* user feedback for liking/unliking etc.

## Functionality Link ##
* displays a list of following users either by clicking on it or by long-tapping on the button (in the small screen version)

## Notifications ##
* when a project is liked the project members get a user notification
* as well as an e-mail notification

## Improvements ##
(these were added to the FollowButton as well)
* the number of followers/likes is now updated after clicking the Like-/FollowButton
* the list of followers/likes is updated after clicking the Like-/FollowButton as well
* a function to handle the states setter functions is passed down instead of the setter functions 

## Before landing

1. PR has meaningful title
1. `yarn lint` passes
1. `yarn format` passes

### CODE REVIEW

**Review by positiveimpact — APPROVED** [review]
LGTM other than my suggestions! :) Great job.

File: frontend/pages/projects/[projectId].js [hunk 1]
```diff
@@ -44,21 +45,54 @@ export async function getServerSideProps(ctx) {
...
+    if (updateCount) {
+      if (userFollows) {
+        setNumberOfFollowers(numberOfFollowers + 1);
+      } else {
+        setNumberOfFollowers(numberOfFollowers - 1);
+      }
+    }
+    setFollowingChangePending(pending);
+  };
+
```
[the remark this answers is not in the document]
Comment (verena-e-p) [reply frontend/pages/projects/[projectId].js:77]: :+1:

File: backend/organization/views/project_views.py [hunk 2]
```diff
@@ -835,6 +883,16 @@ def get_queryset(self):
...
 
+class ListProjectLikesView(ListAPIView):
+    permission_classes = [IsAuthenticated]
+    serializer_class = ProjectLikeSerializer
+
+    def get_queryset(self):
+        project = Project.objects.filter(url_slug=self.kwargs['url_slug'])
+        if not project.exists():
+            return None
+        likes = ProjectLike.objects.filter(project=project[0])
```
[the remark this answers is not in the document]
Comment (verena-e-p) [reply backend/organization/views/project_views.py:894]: Alles klar :)

File: backend/organization/utility/email.py [hunk 3]
```diff
@@ -199,3 +199,30 @@ def send_project_follower_email(user, project_follower, notification):
...
         notification=notification
     )
+
+def send_project_like_email(user, project_like, notification):
+    lang_code = get_user_lang_code(user)
+    liking_user_name = project_like.user.first_name + \
+        " " + project_like.user.last_name
+    subjects_by_language = {
+        "en": "{} liked your project on Climate Connect".format(liking_user_name),
+        "de": "{} folgt jetzt deinem Projekt auf Climate Connect".format(liking_user_name)
```
[the remark this answers is not in the document]
Comment (verena-e-p) [reply backend/organization/utility/email.py:209]: thanks for catching that!

**Pull request merged**